src/read.c

```diff
@@ -156,6 +156,34 @@ char *buf;
     return erode_obj_text(tshirt, buf);
 }
 
+char *
+hawaiian_design(shirt, buf)
+struct obj *shirt;
+char *buf;
+{
+    unsigned style, bg;
+    static const char *hawaiian_styles[] = {
+        "flamingos", "parrots", "toucans", "birds of paradise", /* birds */
```

Comment:
Any interest in making the specific elements more "Nethack-y", like the regular T-shirt messages? Krakens instead of sea turtles, harps instead of ukuleles, silver (dragon) background instead of purple, that sort of thing?

Reply:
I do like this idea, which is why "giant eels" is already in there, but I want to make sure the motifs that are included have a sort of tropical feel that makes the description seem like something you might believably see on a Hawaiian shirt.  Piranhas would probably fit well too, come to think of it.

That effect could also be achieved by adding more verbiage to the monster names, e.g. something like "hula-dancing water nymphs", but the description length is a concern since the whole thing is printed in the end-of-game inventory disclosure list. With this patch applied, a Hawaiian shirt already has a much longer description than most other items, and I'm reluctant to use motifs or backgrounds that will make it even longer. To be honest, I have been debating removing the backgrounds entirely for that reason (to reduce the length), even though I like the way they add to the perceived loudness of the shirt.

Reply:
I included some more NetHack monsters in the commit I just added, which was mostly intended to address the other problem I was talking about (wordiness in end-game disclosure list).